### DISCUSSION TITLE
Difficoltà di collegamento a Home Assistant da remoto

### DISCUSSION THREAD

**Discussion by Newreader** [indomus/forum#3225] [HUB: Home Assistant]
Salve a tutta la comunità,
Ho Home Assistant Core con Docker aggiornato alla versione 2024.8.3 installato su un Raspberry Pi 3 Model B+, uso il reverse proxy Caddy montato sempre su container Docker. Tutto il sistema ha funzionato bene per molto tempo. Anche la rete e i dispositivi ad essa connessi funzionano bene e sono in ogni momento raggiungibili.
Da un paio di giorni, senza aver fatto alcunché, non riesco più a connettermi in remoto con Home Assistant né dal Pc né da due smartphone Android. O almeno ci riesco molto saltuariamente (e già questo è strano). Sul secondo smartphone Android ho provato a disinstallare HA Companion e reinstallarlo e solo all’ennesimo tentativo sono riuscito a collegarmi per poi scollegarmi più tardi.
Invece riesco con tutti i dispositivi sopra menzionati a connettermi in locale.
I log che il sistema mi ha restituito sono:

Unable to fetch auth providers. https://xxxxxxxx.duckdns.org:9000/?auth_callback=1

Login attempt or request with invalid authentication from host-YY-YYY-YY-YYY.retail.telecomitalia.it (YY-YYY-YY-YYY). Requested URL: '/api/websocket'. (Mozilla/5.0 (Linux; Android 14; SM-G991B Build/UP1A.231005.007; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/128.0.6613.128 Mobile Safari/537.36 Home Assistant/2024.7.3-13278 (Android 14; SM-G991B))

Login attempt or request with invalid authentication from host- YY-YYY-YY-YYY.retail.telecomitalia.it (YY-YYY-YY-YYY). Requested URL: '/auth/token'. (Home Assistant/2024.7.3-13278 (Android 14; SM-G991B))Login attempt or request with invalid authentication from host- YY-YYY-YY-YYY.retail.telecomitalia.it (YY-YYY-YY-YYY). Requested URL: '/auth/token'. (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36)

Dai log ho pensato che fosse un problema di Caddy anche perché riesco a connettermi a duckdns,org e al mio sottodominio, per cui ho cancellato e ricreato il container Caddy con i comandi seguenti:
docker container stop caddy
docker container rm caddy
docker run --init -d --name="caddy" -e "TZ=Europe/Rome" -v /home/pi/caddy/Caddyfile:/etc/caddy/Caddyfile -v /home/pi/caddy/data:/data --restart always -p 443:443 -p 9000:9000 caddy:latest

Comandi che sono andati a buon fine tant'è che il reverse porta lo status  Up e le porte in gioco sono quelle canoniche:  80/tcp, 0.0.0.0:443->443/tcp, 2019/tcp, 443/udp, 0.0.0.0:9000->9000/tcp

Il file di configurazione di Caddy è il seguente:
```(https_header) {
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}

https://xxxxxxxxxxx.duckdns.org {
  respond "Questo messaggio è stato modificato dopo che HA Core è passato su Docker e reinstallato di nuovo"
  file_server
}
https://xxxxxxxxxxx.duckdns.org:9000 {
  import https_header
  reverse_proxy http://192.168.yyy.yyy:8123
  @ws: {
    header Connection "Upgrade"
    header Upgrade websocket
  }
}
```
Mentre il blocco http: in configuration.yaml è:
```http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 172.17.0.0/16
```
Nota a margine: durante la scrittura di questa discussione sono riuscito a connettermi con tutti e tre dispositivi, successivamente però si sono sconnessi e al momento non sono più in grado di accedere da remoto con nessuno di essi.
Ho cercato anche in rete le soluzioni e ho visto che anche altri hanno questo problema ma di soluzioni non se ne parla.
Tutto ciò premesso chiedo il vostro aiuto per uscire da questo stato di impasse ringraziandovi anticipatamente.

**Comment by snakuzzo** [comment]
Ciao.
Innanzi tutto...
- disinstallare e reinstallare app non ha alcun senso.
- cancellare e ricreare il container non serve a niente e, soprattutto come hai fatto tu, non ha alcun senso, anche perché lo fai con le stesse configurazioni prima 
- il fatto che funzioni tutto in locale non è assolutamente legato al tuo problema
- il fatto che una volta funzioni e la volta dopo no, l'hai detto pure tu...troppo strano e forse non è un qualcosa legato al reverse proxy.

Dai un'occhiata QUI -> https://community.home-assistant.io/t/login-attempt-or-request-with-invalid-authentication-when-trying-to-access-remotely/373848/26

Soprattutto per quanto riguarda il discorso password e ip bans.

**Reply by Newreader** [reply]
Grazie per la risposta. Ho letto l link che mi hai inviato. Per quanto riguarda ip_bans.yaml non è un file che ho perché non ho il riempimento automatico delle credenziali d'accesso.
Per quanto riguarda la risposta data da Jett, sinceramente non ho capito cosa ha fatto, se non scollegarsi dalle app e dal Browser web. Non ho un tentativo di accesso da parte di un IP sconosciuto ma il blocco viene dall'IP di Telecom Italia.
 Se poi ti riferivi ad altro fammi cortesemente sapere, invitandoti, se possibile, a sostenermi per risolvere il problema.

**Reply by snakuzzo** [reply]
> Grazie per la risposta. Ho letto l link che mi hai inviato. Per quanto riguarda ip_bans.yaml non è un file che ho perché non ho il riempimento automatico delle credenziali d'accesso.

io non ho mai detto che le due cose sono collegate. Rileggi

**Reply by Newreader** [reply]
"Soprattutto per quanto riguarda il discorso password e ip bans."
Scusa ma non capisco la tua risposta. Quali due cose non sono collegate? Se ti riferisci alla risposta di martin_h e a quella di Jett, le cose non sono collegate. Scusami, ma se puoi, sii più esplicito.

**Reply by snakuzzo** [reply]
Ti ho semplicemente indicato di leggere quel topic e di vedere cosa dicono quando parlano di ip ban e password. Non ti ho mai detto che le due cose sono strettamente legate. Più esplicito di dirti leggi, prova quello che dicono e vedi se risolvi...non so cosa dirti.

**Reply by Newreader** [reply]
Sicuramente sono io che non riesco a farmi capire. ho letto il Topic e ho provato le soluzioni, ma non ho risolto per le motivazioni che ti ho scritto prima. Ecco volevo sapere se tu hai altre soluzioni da indicarmi oppure altre possibilità di indagine del sistema che ha sempre funzionato bene nel tempo (anni) fino alla presentazione improvvisa di questa problematica la cui soluzione è fondamentale per la connessione da remoto. Comunque rileggerò più attentamente il Topic e vedrò se ci sono indicate. altre soluzioni possibili.
Quando con Chrome mi connetto in locale non mi chiede né Nome utente né password

Alcune volte mi ha dato anche questo tipo di errore:
Login attempt or request with invalid authentication from rm13.cache.google.com (85.18.30.6). See the log for details.

**Reply by Newreader** [reply]
Ho risolto il problema che, per completezza di informazione, non aveva niente a che vedere con le casistiche suggerite.
Grazie comunque a quanti per spirito di collaborazione hanno voluto aiutarmi a trovare una soluzione al mio problema.

**Reply by snakuzzo** [reply]
Qual era il problema?
Chiedo soprattutto per correttezza di chi potrebbe aver bisogno d'aiuto in futuro per problemi simili.
Faccio presente che questo di fatto è un forum, non un servizio di assistenza dove poi chiudo il ticket e "saluti e baci"

**Reply by Newreader** [reply]
Chiedo scusa se rispondo con ritardo ma non ho avuto la possibilità di farlo prima. Faccio presente, per mia etica personale, che non è mai stata mia intenzione, laddove ne avessi la possibilità, non adoperarmi per la soluzione degli altri problemi o fornire soluzioni in merito alle mie esperienze personali.
Fatta questa doverosa premessa, almeno nel mio caso il problema nasceva da uno Shelly 2.5 che adoperavo per gestire una tapparella che, probabilmente per il cattivo funzionamento del condensatore (tant'è che avvicinandomi sentivo il classico sibilo), si connetteva e disconnetteva in continuazione provocando il problema di cui ho scritto.
Una volta sostituito, alcuni giorni fa, lo Shelly 2.5 PM con uno Shelly 2 PM Plus  (ma sarebbe stata la cosa sostituirlo con un altro identico) e aver preso preso il tempo per una seria verifica del risultato, il problema non si è più ripresentato.

**Reply by snakuzzo** [reply]
Questa che indichi come soluzione non ha alcun nesso con il problema che hai indicato all"inizio.

**Reply by Newreader** [reply]
Questa è stata la mia esperienza con risultati visibili, ad oggi il problema non si è più ripresentato. Oppure il sistema si è "sistemato" in qualche modo da solo. Non ho fatto alcun intervento aggiuntivo né hardware né software.

**Reply by snakuzzo** [reply]
Guarda, penso più la seconda...e comunque non è che vive di vita propria. Sarà successo qualcosa, al di fuori della tua conoscenza, che non ti dà più quel problema...ma lo shelly non c'entra assolutamente nulla con la mancata connettività da remoto, lo capirebbe anche un bambino di 5 anni.

**Reply by Newreader** [reply]
Se fosse possibile evitare commenti banali e superflui te ne sarei grato. Ho riportato solo la mia esperienza rilevando la concomitanza degli eventi, anche se indipendenti.

**Reply by snakuzzo** [reply]
Non evito alcun commento, perché non è superfluo.
Facevi meglio a scrivere "il problema si è risolto da solo, non so per quale motivo ora non ho più il problema", ma di sicuro non che cambiando lo shelly si risolve un problema di connettività...perché, oltre a dimostrare che non hai idea di cosa stai facendo, dai anche informazioni sbagliate e fuorvianti ad altra gente che viene sul forum in cerca di soluzioni alle proprie problematiche.

Questa discussione in teoria potrebbe anche essere anche eliminata, visto che non dà alcun valore aggiunto

**Reply by Newreader** [reply]
Premesso che se vuoi puoi anche eliminare la discussione, anzi sono io che cortesemente te lo chiedo. Torno a ripetere che è stata solo e unicamente la mia esperienza che non ha mai avuto la pretesa di dare informazioni (sbagliate o fuorvianti che siano). Che poi cambiando lo shelly non si risolve il problema della connettività ci sta e ne sono consapevole anche perché in locale la connessione funzionava. Ma la mia esperienza è stata quella e così l'ho descritta. Per quanto riguarda il tuo commento mi riferivo non al contenuto in sé per sé stesso ma alla frase "lo capirebbe anche un bambino di 5 anni" che poteva essere evitata.
Dato che tengo troppo a far parte di questa comunità e per evitare fraintendimenti concordo nell'affermare che "il problema si è risolto da solo, non so per quale motivo ora non ho più il problema".
Riconosco la tua competenza e ti stimo per quello che fai per aiutare gli altri ma forse (questa è una mia opinione) alcune volte un po' di sarcasmo in meno non guasterebbe.

**Reply by snakuzzo** [reply]
Ok. Ciao.
